kernel-python/urth/widgets/urth_widget.py

```diff
@@ -14,4 +15,58 @@ def send_state(self, key=None):
         """Overrides the Widget send_state to prevent
         an unnecessary initial state message.
         """
-        pass
\ No newline at end of file
+        pass
+
+    def _send_update(self, attribute, value):
+        """
+        Sends a message to update the front-end state of the given attribute.
+        """
+        msg = {
+            "method": "update",
+            "state": {
+                attribute: value
+            }
+        }
+        self._send(msg)
+
+    def send_status(self, status, msg=""):
+        """
+        Sends a message to inform the front-end of the execution status.
+
+        Parameters
+        ----------
+        status : string
+            "ok" for success, "error" for failure.
+        msg : string
+            Message accompanying the status, e.g. an error message.
+        """
+        self._send({
+            "method": "update",
+            "state": {
+                "status": {
```

Comment:
As above... consider using `__status__` to avoid taking over that variable name. Also investigate using a message other then 'update'.